internal/assert/assertion_mongo.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (C) MongoDB, Inc. 2023-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+// assertion_mongo.go contains MongoDB-specific extensions to the "assert"
+// package.
+
+package assert
+
+import (
+	"fmt"
+	"reflect"
+	"unsafe"
+)
+
+// DifferentAddressRanges asserts that two byte slices reference distinct memory
+// address ranges, meaning they reference different underlying byte arrays.
+func DifferentAddressRanges(t TestingT, a, b []byte) {
```

Comment:
It doesn't have to be done in this PR, but it would be helpful to add a test for this assertion case. It could be a greener build ticket.

Reply:
Good idea. I'll look at adding either a test or a ticket.